### PR TITLE
Prevent logging in successful unit tests #601

### LINKED ISSUE #601: Prevent logging / warning during unit tests
**Description**:
Following recent package updates, we now get more errors/warnings/logging that doesn't fail unit tests, but can clog the terminal and make it harder to debug legitimate failures. 

**Acceptance criteria**:
- [ ] No logging for passing unit tests

### PR DESCRIPTION
**Pull request by patrick-austin**
## Description

- Add `__mocks__/react-i18next.jsx` to common so we can mock translations for refactored cardview components
- Explicitly mock `console.error` in one of the cardView tests. The error is for using a dynamic variable to set whether an Accordion (aka ExpansionPanel) is expanded by default. I've had a look at how these panels are controlled, and I don't think we want to change our functionality here, so just surpressed the warning in the tests.
- Add `item` to `Grid` in cardView (no functional change expected)
- Treat id checks as asynchronous in `pageContainer` tests like we do in `pageRouting`
- In lots of tests, move mocked `axios.get` into `beforeEach` and/or add in mocking for `axios.post` and `axios.delete` (corresponding to adding and removing to/from cart) to prevent unhandled request logging
- Reduce size of `searchPageTable` snapshot by using `shallow` instead of `mount`
- Change how we treat disabled tabs in search, instead of removing them and using `none` as a placeholding (which leads to errors about having a value that doesn't correspond to a tab) use a `Tab` set to `display: 'none'`. This should achieve the same functionality as before but without the warnings.
 
## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] Check no logging in successful unit tests
- [ ] Check functionality of tabs in search is unaffected: unchecking an entity and researching should remove that tab from the UI and the default table dispalyed should be the first of the three entities that's checked.

## Agile board tracking
closes #601

### CODE REVIEW

File: packages/datagateway-common/src/card/cardView.component.tsx [hunk 1]
```diff
@@ -760,7 +760,7 @@ const CardView = (props: CardViewProps): React.ReactElement => {
                 })}
               </List>
             ) : (
-              <Grid xs={12} md={8}>
+              <Grid item xs={12} md={8}>
```
Comment (sam-glendenning) [inline packages/datagateway-common/src/card/cardView.component.tsx:763]: I assume `item` is always going to be true, else this code will not work. Safety alternative is `item={item}`

File: packages/datagateway-search/src/searchPageTable.test.tsx [hunk 2]
```diff
@@ -5,26 +5,33 @@ import { MemoryRouter } from 'react-router';
...
+import axios from 'axios';
 
 describe('SearchPageTable', () => {
   let mount: typeof enzymeMount;
+  let shallow: typeof enzymeShallow;
   let state: StateType;
 
   beforeEach(() => {
     mount = createMount();
+    shallow = createShallow({ untilSelector: 'Paper' });
```
Comment (sam-glendenning) [inline packages/datagateway-search/src/searchPageTable.test.tsx:24]: Is this shallow snapshot being rendered to the correct selector? As I'm unsure of its exact function, could use clarification from Louise/Victor.

**Review by sam-glendenning — COMMENTED** [review]
Overall this change looks good to me and not noticing any unnecessary output in terminal when running tests. From a code point of view it looks fine for the most part, just added a couple of comments in files for clarification. 

`datagateway-search/src/table/investigationSearchTable.component.test.tsx` still makes reference to `instrumentId` and `facilityCycleId` on one occasion when rendering an `<InvestigationSearchTable />` and I'm not entirely sure if it is intended or not (in other parts of the file, these attributes are removed). 

To summarise, it seems good to me but I won't approve yet because I would like a final once-over from someone on Monday (likely either Louise or Victor) to make sure everything is correct as I'm still unclear on the overarching functionality of some of these tests. I'll organise this on Monday.

File: packages/datagateway-common/src/card/cardView.component.tsx (hunk 1 above)
Comment (patrick-austin) [reply packages/datagateway-common/src/card/cardView.component.tsx:763]: `item` is a boolean prop for the MUI `Grid` component which defaults to false. This is shorthand for `item={True}`, you can see the same thing elsewhere with the `container` prop.

https://material-ui.com/api/grid/#props

File: packages/datagateway-search/src/searchPageTable.test.tsx (hunk 2 above)
Comment (patrick-austin) [reply packages/datagateway-search/src/searchPageTable.test.tsx:24]: Can't really remember this one, but from looking at the snapshot for this test rendering until `Paper` stops the individual `InvestigationSearchTable`, `DatasetSearchTable`, `DatafileSearchTable` from rendering. Maybe these were causing some excess logging, or maybe it was just to make the snapshot smaller? If there was a reason to have them in the snapshot it could be reverted or just make sure it's being covered by the tests that actaully mount.

**Comment by patrick-austin** [comment]
> 
> 
> 
> `datagateway-search/src/table/investigationSearchTable.component.test.tsx` still makes reference to `instrumentId` and `facilityCycleId` on one occasion when rendering an `<InvestigationSearchTable />` and I'm not entirely sure if it is intended or not (in other parts of the file, these attributes are removed).

At a guess I'd say that having them only caused errors when we fully mounted, so I removed all of those but missed the `'renders details panel correctly'` because it's shallow and so doesn't cause errors, is near the bottom of the file and if I was just find/replacing `instrumentId="4" facilityCycleId="5"` the fact it has a newline would mean it wouldn't show up.

Ultimately it shouldn't have them, as these only apply to `ISISInvestigationsTable` not `InvestigationSearchTable`, but it doesn't look like it's logging any warnings by leaving them in.

File: packages/datagateway-common/src/card/cardView.component.tsx (hunk 1 above)
Comment (sam-glendenning) [reply packages/datagateway-common/src/card/cardView.component.tsx:763]: I understand the shorthand, just needed to know it would be true in all circumstances. I'll give it a tick.

File: packages/datagateway-search/src/searchPageTable.test.tsx (hunk 2 above)
Comment (VKTB) [reply packages/datagateway-search/src/searchPageTable.test.tsx:24]: Can't see any logging when changing it back to `mount` however the snapshot gets bigger as `InvestigationSearchTable`, `DatasetSearchTable`, `DatafileSearchTable` are rendered. Looking at the coverage, all lines are covered when using `shallow` so I would just leave it as it is.

**Comment by VKTB** [comment]
> > `datagateway-search/src/table/investigationSearchTable.component.test.tsx` still makes reference to `instrumentId` and `facilityCycleId` on one occasion when rendering an `<InvestigationSearchTable />` and I'm not entirely sure if it is intended or not (in other parts of the file, these attributes are removed).
> 
> At a guess I'd say that having them only caused errors when we fully mounted, so I removed all of those but missed the `'renders details panel correctly'` because it's shallow and so doesn't cause errors, is near the bottom of the file and if I was just find/replacing `instrumentId="4" facilityCycleId="5"` the fact it has a newline would mean it wouldn't show up.
> 
> Ultimately it shouldn't have them, as these only apply to `ISISInvestigationsTable` not `InvestigationSearchTable`, but it doesn't look like it's logging any warnings by leaving them in.

@sam-glendenning As @patrick-austin says, these references do not cause any issues but please feel free to remove them in case you think they can cause confusion in future. The rest of the changes look good to me.

**Review by sam-glendenning — APPROVED** [review]
Taken out those unnecessary props (prevents confusion later and keeps things clean and tidy) and all tests are happy. All looks good to me.
